### DISCUSSION TITLE
Event for detecting Asset File Change

### DISCUSSION THREAD

**Discussion by lukasIO** [craftcms/cms#9717] [Q&A] [answered]
Hi,
 
is there a way to detect when the file of an asset has been changed? 
Which means, I would like to ignore when other fields (title etc.) have been changed and only react to either: Upload, editImage, replace

I was looking at `Asset::EVENT_AFTER_SAVE` but unfortunately that event is (understandably) being triggered each time the asset's field value changes/save button is pressed and on top of that it is triggered once for each site.

I also found `Assets::EVENT_AFTER_REPLACE_ASSET` which gets me half-way there, but I couldn't figure out how to listen to the initial upload/and or image edits.

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
You can check the asset’s scenario from `EVENT_AFTER_SAVE` to get an idea of what’s happening with the file.

```php
use craft\elements\Asset;
use craft\events\ModelEvent;
use yii\base\Event;

Event::on(Asset::class, Asset::EVENT_AFTER_SAVE, function(ModelEvent $event) {
    /** @var Asset $asset */
    $asset = $event->sender;
    $scenario = $asset->getScenario();

    if (in_array($asset->getScenario(), [
        Asset::SCENARIO_CREATE,
        Asset::SCENARIO_MOVE,
        Asset::SCENARIO_REPLACE,
        Asset::SCENARIO_FILEOPS,
    ])) {
        // something is happening with the actual file
    }
});
```
